### PR TITLE
Make send check non-blocking to allow other workers to be progressed

### PR DESCRIPTION
**Pull request by tvegas1**
## What
Make `ucx_send_check()` connection establishment non-blocking. This prepares later patch changing worker allocation scheme.

## Why
When blocking on send check, this could cause dead-locks if on more than one node, other endpoints on other workers need to be progressed.

## How
Make send check non-blocking. Next cleanup step can be moving send/recv check functions to connect/accept, delayed as it is causing other issues (#128)

### CODE REVIEW

File: src/ucx_plugin.c [hunk 1]
```diff
@@ -597,41 +597,62 @@ static ncclResult_t ucx_send_check(ucx_comm_t *comm) {
...
+  comm->connect_req = ucp_req;
+
+out_check_status:
+  status = ucp_request_check_status(comm->connect_req);
+  if (status == UCS_INPROGRESS) {
+    return ncclSuccess;
   }
 
+  if (status != UCS_OK) {
+    WARN("Send check requested returned error (%s)", ucs_status_string(status));
```
Comment (brminich) [inline src/ucx_plugin.c:642]: seems we need to free `comm->msg` here

File: src/ucx_plugin.c [hunk 2]
```diff
@@ -597,41 +597,62 @@ static ncclResult_t ucx_send_check(ucx_comm_t *comm) {
...
-    ucp_request_free(ucp_req);
+  } else if (ucp_req == NULL) {
+    goto out_set_ready;
+  }
+
+  assert(comm->connect_req == NULL);
+  comm->connect_req = ucp_req;
+
+out_check_status:
+  status = ucp_request_check_status(comm->connect_req);
```
Comment (brminich) [inline src/ucx_plugin.c:636]: i'd poke a progress here. Or where is it invoked?

Comment (tvegas1) [reply src/ucx_plugin.c:636]: it is invoked at the start of the function so we should be good i think

File: src/ucx_plugin.c (hunk 1 above)
Comment (tvegas1) [reply src/ucx_plugin.c:642]: fixed

File: src/ucx_plugin.c (hunk 2 above)
Comment (brminich) [reply src/ucx_plugin.c:637]: right, i missed that

**Review by brminich — APPROVED** [review]

**Review by bureddy — APPROVED** [review]

**Pull request merged**